test/export/mcap_test.py

```diff
@@ -0,0 +1,221 @@
+        pose = [1.0, 2.0, 3.0, 0.12, 0.24, 0.36, 0.8]
+        dut = MCAP(simulation)
+        dut._add_agent_entity_to_scene(agent_name, pose, scene_update, sim_time)
+        self.assertEqual(len(scene_update.entities), 1)
+        agent_entity = scene_update.entities[0]
+        self.assertEqual(agent_entity.id, agent_name)
+        self.assertEqual(agent_entity.timestamp.ToNanoseconds(), sim_time)
+        self.assertEqual(agent_entity.frame_id, dut._ROOT_FRAME_ID)
+        self.assertEqual(len(agent_entity.cubes), 1)
+        agent_cube = agent_entity.cubes[0]
+        self.assertEqual(agent_cube.size, dut._AGENT_BOX_SIZE)
+        self.assertEqual(agent_cube.color, dut._AGENT_COLOR)
+        pose = Pose(
+            position=Vector3(x=pose[0], y=pose[1], z=pose[2]),
+            orientation=Quaternion(
+                x=pose[3], y=pose[4], z=pose[5], w=pose[6])
+        )
+        self.assertEqual(agent_cube.pose, pose)
+
+
+class TestMCAPDumpMethod(unittest.TestCase):
+    _FILE_NAME = "mcap_test.mcap"
+
+    def setUp(self) -> None:
+        sim = GetSimulationMock([GetSimulationState(0.), GetSimulationState(
+            1.), GetSimulationState(2.)], GetRoadNetwork())
+        self.tmp_dir = mkdtemp()
+        self.obj_file_path = os.path.join(self.tmp_dir, self._FILE_NAME)
+        self.dut = MCAP(sim, self.obj_file_path)
+        return super().setUp()
+
+    def tearDown(self) -> None:
+        shutil.rmtree(self.tmp_dir)
+
+    def test_dump_method_default_values(self):
+        self.dut.dump()
+        self.assertTrue(os.path.exists(self.obj_file_path))
+        with open(self.obj_file_path, "rb") as f:
+            reader = make_reader(f, decoder_factories=[DecoderFactory()])
+            scene_update_msgs = list(
+                reader.iter_decoded_messages(topics=["/scene_update"]))
+            self.assertEqual(len(scene_update_msgs), 3)
+            for schema, channel, message, proto_msg in scene_update_msgs:
+                self.assertEqual(schema.name, "foxglove.SceneUpdate")
+                # TODO: Check the content of the message.
+
+            pose_msgs = list(reader.iter_decoded_messages(
+                topics=["/agent_1/pose"]))
+            # Pose msg isn't being published in first iteration.
+            self.assertEqual(len(pose_msgs), 2)
+            for schema, channel, message, proto_msg in pose_msgs:
+                self.assertEqual(schema.name, "foxglove.Pose")
+                # TODO: Check the content of the message.
+
+    def test_dump_method_exporting_signals(self):
+        self.dut.dump(export_signals=['signal_1'])
+        self.assertTrue(os.path.exists(self.obj_file_path))
+        with open(self.obj_file_path, "rb") as f:
+            reader = make_reader(f, decoder_factories=[DecoderFactory()])
+            behavior_msgs = list(reader.iter_decoded_messages(
+                topics=["/agent_1/behavior/0/signal_1"]))
+            # behavior msg isn't being published in first iteration.
```

Comment:
This is the current behavior, perhaps we should create an issue to also publish the initial states in iteration zero.

Reply:
https://github.com/maliput/maliput_sim/issues/43

Reply:
I think we should, yes.